pkg/url/url.go

```diff
@@ -1086,3 +1125,19 @@ func Push(client *occlient.Client, kClient *kclient.Client, parameters PushParam
 
 	return nil
 }
+
+// UpdateEndpointsInDevfile writes the provided endpoint information into devfile
+func UpdateEndpointsInDevfile(devObj parser.DevfileObj, containerEndpointMap map[string]map[string]parsercommon.Endpoint) error {
+	components := devObj.Data.GetComponents()
+	for _, component := range components {
+		if component.Container != nil {
+			var endpoints []common.Endpoint
+			for _, endpoint := range containerEndpointMap[component.Container.Name] {
+				endpoints = append(endpoints, endpoint)
+			}
```

Comment:
Why not just pass the container name and endpoint and use it for matching and appending instead of looping? 

I know it won't work for delete but we can have a separate delete function which just takes the url name.

Reply:
It's because initially I want to support overriding the existing endpoint name in devfile.yaml, then this function need to support both deletion and creation. 
Since we do not want to provide overwriting of existed endpoint name. it can be separated into two separate functions for add & remove respectively.

Reply:
updated